packages/@react-aria/interactions/src/useFocusWithin.ts

```diff
@@ -70,7 +71,7 @@ export function useFocusWithin(props: FocusWithinProps): FocusWithinResult {
   let onFocus = useCallback((e: FocusEvent) => {
     // Double check that document.activeElement actually matches e.target in case a previously chained
     // focus handler already moved focus somewhere else.
-    if (!state.current.isFocusWithin && document.activeElement === e.target) {
+    if (!state.current.isFocusWithin && getDeepActiveElement() === e.target) {
```

Comment:
down below you say that e.target will be the shadowRoot, but won't getDeepActiveElement find the element inside the shadowRoot?